src/Datagrid/Pager.php

```diff
@@ -38,12 +40,32 @@ public function computeNbResult()
             $countQuery->setParameters($this->getParameters());
         }
 
-        $countQuery->select(sprintf(
-            'count(%s %s.%s) as cnt',
-            $countQuery instanceof ProxyQuery && !$countQuery->isDistinct() ? null : 'DISTINCT',
-            current($countQuery->getRootAliases()),
-            current($this->getCountColumn())
-        ));
+        if (\count($this->getCountColumn()) > 1) {
```

Comment:
Can you please try to split this method

Reply:
I split code of build count query for composite and single primary keys